src/Mod/Sketcher/Gui/CommandConstraints.cpp

```diff
@@ -2336,6 +2336,35 @@ void CmdSketcherConstrainDistance::activated(int iMsg)
             else
                 finishDatumConstraint (this, Obj, true);
 
+            return;
+        } else if (geom1->getTypeId() == Part::GeomCircle::getClassTypeId()
+        && geom2->getTypeId() == Part::GeomLineSegment::getClassTypeId() ) { // circle to line distance
+            auto circleSeg = static_cast<const Part::GeomCircle*>(geom1);
```

Comment:
Maybe I missed something, but it appears that you are not checking a case where geom1 is the line and geom2 is the circle (so the user clicked in the other order).

If it is not handled, look around for some checking and a call to std::swap, so that you are sure geom1 is the circle when arriving here.